### PR TITLE
ENG 256 fix gosec issues

### PR DESCRIPTION
**Pull request by danburck**
Part of https://linear.app/tharsis/issue/ENG-256/fix-gosec-issues-on-ethermint-and-evmos

## Description

This PR fixes the last vulerabilities pointed out by `gosec`  in order to make `gosec` required for github workflows.

## Note

On merge, an admin of this repo needs to set the gosec github action to required

### CODE REVIEW

File: rpc/ethereum/namespaces/debug/api.go [hunk 1]
```diff
@@ -351,7 +351,10 @@ func (a *API) StartCPUProfile(file string) error {
 		}
 		if err := pprof.StartCPUProfile(f); err != nil {
 			a.logger.Debug("cpu profiling already in use", "error", err.Error())
-			f.Close()
+			if err := f.Close(); err != nil {
+				a.logger.Debug("failed to close cpu profile file")
+				return errors.New("failed to close cpu profile file")
```
Comment (fedekunze) [inline rpc/ethereum/namespaces/debug/api.go:356]: can you wrap the error?
```suggestion
				return errors.Wrap(err, "failed to close cpu profile file")
```

File: rpc/ethereum/namespaces/debug/api.go [hunk 2]
```diff
@@ -375,7 +378,10 @@ func (a *API) StopCPUProfile() error {
 	case a.handler.cpuFile != nil:
 		a.logger.Info("Done writing CPU profile", "profile", a.handler.cpuFilename)
 		pprof.StopCPUProfile()
-		a.handler.cpuFile.Close()
+		if err := a.handler.cpuFile.Close(); err != nil {
+			a.logger.Debug("failed to close cpu file")
+			return errors.New("failed to close cpu file")
```
Comment (fedekunze) [inline rpc/ethereum/namespaces/debug/api.go:383]:
```suggestion
			return errors.Wrap(err, "failed to close cpu file")
```

File: rpc/ethereum/namespaces/debug/trace.go [hunk 3]
```diff
@@ -47,7 +47,11 @@ func (a *API) StartGoTrace(file string) error {
 	}
 	if err := trace.Start(f); err != nil {
 		a.logger.Debug("Go tracing already started", "error", err.Error())
-		f.Close()
+		if err := f.Close(); err != nil {
+			a.logger.Debug("failed to close trace file")
+			return errors.New("failed to close trace file")
```
Comment (fedekunze) [inline rpc/ethereum/namespaces/debug/trace.go:52]:
```suggestion
			return errors.Wrap(err, "failed to close trace file")
```

File: rpc/ethereum/namespaces/debug/trace.go [hunk 4]
```diff
@@ -68,7 +72,10 @@ func (a *API) StopGoTrace() error {
 		return errors.New("trace not in progress")
 	}
 	a.logger.Info("Done writing Go trace", "dump", a.handler.traceFilename)
-	a.handler.traceFile.Close()
+	if err := a.handler.traceFile.Close(); err != nil {
+		a.logger.Debug("failed to close trace file")
+		return errors.New("failed to close trace file")
```
Comment (fedekunze) [inline rpc/ethereum/namespaces/debug/trace.go:77]:
```suggestion
		return errors.Wrap(err, "failed to close trace file")
```

File: server/start.go [hunk 5]
```diff
@@ -244,7 +244,9 @@ func startInProcess(ctx *server.Context, clientCtx client.Context, appCreator ty
 		cpuProfileCleanup = func() {
 			ctx.Logger.Info("stopping CPU profiler", "profile", cpuProfile)
 			pprof.StopCPUProfile()
-			f.Close()
+			if err := f.Close(); err != nil {
```
Comment (fedekunze) [inline server/start.go:247]: return error

File: server/start.go [hunk 6]
```diff
@@ -439,7 +441,9 @@ func startInProcess(ctx *server.Context, clientCtx client.Context, appCreator ty
 		if grpcSrv != nil {
 			grpcSrv.Stop()
 			if grpcWebSrv != nil {
-				grpcWebSrv.Close()
+				if err := grpcWebSrv.Close(); err != nil {
```
Comment (fedekunze) [inline server/start.go:444]: return

File: rpc/ethereum/namespaces/debug/api.go [hunk 7]
```diff
@@ -351,7 +352,10 @@ func (a *API) StartCPUProfile(file string) error {
 		}
 		if err := pprof.StartCPUProfile(f); err != nil {
 			a.logger.Debug("cpu profiling already in use", "error", err.Error())
-			f.Close()
+			if err := f.Close(); err != nil {
+				a.logger.Debug("failed to close cpu profile file")
+				return sdkerrors.Wrap(err, "failed to close cpu profile file")
```
Comment (fedekunze) [inline rpc/ethereum/namespaces/debug/api.go:357]: see comment above

File: rpc/ethereum/namespaces/debug/api.go [hunk 8]
```diff
@@ -375,7 +379,10 @@ func (a *API) StopCPUProfile() error {
 	case a.handler.cpuFile != nil:
 		a.logger.Info("Done writing CPU profile", "profile", a.handler.cpuFilename)
 		pprof.StopCPUProfile()
-		a.handler.cpuFile.Close()
+		if err := a.handler.cpuFile.Close(); err != nil {
+			a.logger.Debug("failed to close cpu file")
+			return sdkerrors.Wrap(err, "failed to close cpu file")
```
Comment (fedekunze) [inline rpc/ethereum/namespaces/debug/api.go:384]: see comment above

File: server/start.go [hunk 9]
```diff
@@ -359,7 +363,7 @@ func startInProcess(ctx *server.Context, clientCtx client.Context, appCreator ty
 		if config.GRPCWeb.Enable {
 			grpcWebSrv, err = servergrpc.StartGRPCWeb(grpcSrv, config.Config)
 			if err != nil {
-				ctx.Logger.Error("failed to start grpc-web http server: ", err)
+				ctx.Logger.Error("failed to start grpc-web http server", "error", err)
```
Comment (fedekunze) [inline server/start.go:366]: nice, thanks for the fix

File: server/start.go [hunk 10]
```diff
@@ -459,6 +466,7 @@ func startInProcess(ctx *server.Context, clientCtx client.Context, appCreator ty
 		}
 
 		logger.Info("Bye!")
+		return nil
 	}()
```
Comment (fedekunze) [inline server/start.go:470]: Ref: https://stackoverflow.com/questions/57740428/handling-errors-in-defer and https://trstringer.com/golang-deferred-function-error-handling/

Comment (danburck) [reply server/start.go:470]: @fedekunze nice, thank you :)

File: rpc/ethereum/namespaces/debug/trace.go [hunk 11]
```diff
@@ -23,6 +23,8 @@ import (
 	"errors"
 	"os"
 	"runtime/trace"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
```
Comment (fedekunze) [inline rpc/ethereum/namespaces/debug/trace.go:27]:
```suggestion
	stderrors "github.com/pkg/errors"
```

File: rpc/ethereum/namespaces/debug/trace.go [hunk 12]
```diff
@@ -47,7 +49,11 @@ func (a *API) StartGoTrace(file string) error {
 	}
 	if err := trace.Start(f); err != nil {
 		a.logger.Debug("Go tracing already started", "error", err.Error())
-		f.Close()
+		if err := f.Close(); err != nil {
+			a.logger.Debug("failed to close trace file")
+			return sdkerrors.Wrap(err, "failed to close trace file")
```
Comment (fedekunze) [inline rpc/ethereum/namespaces/debug/trace.go:54]:
```suggestion
			return stderrors.Wrap(err, "failed to close trace file")
```

File: rpc/ethereum/namespaces/debug/trace.go [hunk 13]
```diff
@@ -68,7 +74,10 @@ func (a *API) StopGoTrace() error {
 		return errors.New("trace not in progress")
 	}
 	a.logger.Info("Done writing Go trace", "dump", a.handler.traceFilename)
-	a.handler.traceFile.Close()
+	if err := a.handler.traceFile.Close(); err != nil {
+		a.logger.Debug("failed to close trace file")
+		return sdkerrors.Wrap(err, "failed to close trace file")
```
Comment (fedekunze) [inline rpc/ethereum/namespaces/debug/trace.go:79]:
```suggestion
		return stderrors.Wrap(err, "failed to close trace file")
```

File: server/flags/flags.go [hunk 14]
```diff
@@ -49,15 +49,15 @@ const (
...
-	cmd.PersistentFlags().String(flags.FlagChainID, "testnet", "Specify Chain ID for sending Tx")
-	cmd.PersistentFlags().String(flags.FlagFrom, "", "Name or address of private key with which to sign")
-	cmd.PersistentFlags().String(flags.FlagFees, "", "Fees to pay along with transaction; eg: 10aphoton")
-	cmd.PersistentFlags().String(flags.FlagGasPrices, "", "Gas prices to determine the transaction fee (e.g. 10aphoton)")
-	cmd.PersistentFlags().String(flags.FlagNode, "tcp://localhost:26657", "<host>:<port> to tendermint rpc interface for this chain")
-	cmd.PersistentFlags().Float64(flags.FlagGasAdjustment, flags.DefaultGasAdjustment, "adjustment factor to be multiplied against the estimate returned by the tx simulation; if the gas limit is set manually this flag is ignored ")
-	cmd.PersistentFlags().StringP(flags.FlagBroadcastMode, "b", flags.BroadcastSync, "Transaction broadcasting mode (sync|async|block)")
-	cmd.PersistentFlags().String(flags.FlagKeyringBackend, keyring.BackendOS, "Select keyring's backend")
+func AddTxFlags(cmd *cobra.Command) (*cobra.Command, error) {
+	cmd.Flags().String(flags.FlagChainID, "testnet", "Specify Chain ID for sending Tx")
```
Comment (fedekunze) [inline server/flags/flags.go:53]: why the change to Flags?

Comment (danburck) [reply server/flags/flags.go]: @fedekunze Ok after reading the cobra docs it makes much more sense :) I added the Persistent flags again and decided to remove the required `chain-id` in the rootCmd, cause otherwise all commands would require a `chain-id`.

**Review by fedekunze — APPROVED** [review]
ACK
